x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/three_way_diff/comparison_side/field_comparison_side.tsx

```diff
@@ -7,46 +7,36 @@
 
 import React, { useState } from 'react';
 import { EuiFlexGroup, EuiTitle } from '@elastic/eui';
-import { VersionsPicker } from '../versions_picker/versions_picker';
-import type { Version } from '../versions_picker/constants';
-import { SelectedVersions } from '../versions_picker/constants';
+import { VersionsPicker } from './versions_picker/versions_picker';
+import type { Version } from './versions_picker/constants';
+import { SelectedVersions } from './versions_picker/constants';
+import { FieldUpgradeSideHeader } from '../field_upgrade_side_header';
+import { useFieldUpgradeContext } from '../rule_upgrade/field_upgrade_context';
 import { pickFieldValueForVersion } from './utils';
-import type {
-  DiffableAllFields,
-  ThreeWayDiff,
-} from '../../../../../../../common/api/detection_engine';
 import { getSubfieldChanges } from './get_subfield_changes';
 import { SubfieldChanges } from './subfield_changes';
-import { SideHeader } from '../components/side_header';
 import { ComparisonSideHelpInfo } from './comparison_side_help_info';
 import * as i18n from './translations';
 
-interface ComparisonSideProps<FieldName extends keyof DiffableAllFields> {
-  fieldName: FieldName;
-  fieldThreeWayDiff: ThreeWayDiff<DiffableAllFields[FieldName]>;
-  resolvedValue: DiffableAllFields[FieldName];
-}
+export function FieldComparisonSide(): JSX.Element {
+  const { fieldName, fieldDiff, finalDiffableRule } = useFieldUpgradeContext();
+  const resolvedValue = finalDiffableRule[fieldName];
 
-export function ComparisonSide<FieldName extends keyof DiffableAllFields>({
-  fieldName,
-  fieldThreeWayDiff,
-  resolvedValue,
-}: ComparisonSideProps<FieldName>) {
   const [selectedVersions, setSelectedVersions] = useState<SelectedVersions>(
     SelectedVersions.CurrentFinal
   );
 
   const [oldVersionType, newVersionType] = selectedVersions.split('_') as [Version, Version];
 
-  const oldFieldValue = pickFieldValueForVersion(oldVersionType, fieldThreeWayDiff, resolvedValue);
+  const oldFieldValue = pickFieldValueForVersion(oldVersionType, fieldDiff, resolvedValue);
 
-  const newFieldValue = pickFieldValueForVersion(newVersionType, fieldThreeWayDiff, resolvedValue);
+  const newFieldValue = pickFieldValueForVersion(newVersionType, fieldDiff, resolvedValue);
 
   const subfieldChanges = getSubfieldChanges(fieldName, oldFieldValue, newFieldValue);
 
   return (
     <>
-      <SideHeader>
+      <FieldUpgradeSideHeader>
         <EuiFlexGroup direction="row" alignItems="center">
           <EuiTitle size="xxs">
             <h3>
```

Comment:
I've noticed that `Diff view` doesn't fit on a single line at 1280px.

<img width="1392" alt="Scherm­afbeelding 2024-12-03 om 19 12 48" src="https://github.com/user-attachments/assets/9a023245-58e1-4e2a-8892-89a7da1274d8">

One way to fix it:
```suggestion
            <h3
              css={`
                white-space: nowrap;
              `}
            >
```

Reply:
Maybe we could do the same for field headers - add nowrap to all items except the last one to prevent title and conflict badge taking up two lines and expanding header height. wdyt?

<img width="948" alt="Scherm­afbeelding 2024-12-03 om 19 33 59" src="https://github.com/user-attachments/assets/50d9d737-4ecc-47c3-a0cc-a2fb8ae121c9">

<img width="949" alt="Scherm­afbeelding 2024-12-03 om 19 39 50" src="https://github.com/user-attachments/assets/649c3de8-4c14-4603-9322-a52c235f3004">

Reply:
I think we can make min width of the dropdown a bit wider to fit longer lines. Also, some longer titles still wrap for me.

<img width="1281" alt="Scherm­afbeelding 2024-12-05 om 10 13 49" src="https://github.com/user-attachments/assets/eb05cf1b-a0bc-4da9-8275-70ced34f783e">

Reply:
I specified min-width for versions picker to make English text readable on smaller screens. I'm not sure if should change anything in the header. For example `KQL query` is still readable.